### PR TITLE
Add a timer to ignite/contrib

### PR DESCRIPTION
**Pull request by elanmart**
This PR adds a `Timer` class to `ignite.contrib`. I use it frequently to measure time it takes to process a single batch / run full validation etc.

It's based on the `timer` that was available in `torch`'s `torchnet`. It can be paused and resumed, and can 
compute average time if `unit=True` is passed.

It has an `attach` methods that can attach the timer to the `Engine`, so that you can save few lines of code (not sure if it's a good idea though).

If you think this is something you'd like to merge, I'd have to add
- [x] docs
- [ ] tests
- [x] example

The PR also sorts and removes unused imports.

### CODE REVIEW

**Comment by alykhantejani** [comment]
Hi @elanmart, 

This looks good to me, I think it'll be a useful addition. The only thing I think we should think about is where this (and things like this) live. Right now we have `ignite.handlers` and if we add a `ignite.contrib` which is also a bunch of handlers, what's the distinction?

Maybe we can move what's in `handlers` to `contrib` or vise versa

**Comment by alykhantejani** [comment]
@elanmart shall we move this to `ignite.handlers` for now. Once we sort out metrics and the evaluator we can then clean up handlers and maybe just rename it to contrib?

**Comment by elanmart** [comment]
@alykhantejani yes, sorry for the delay, I'll sort out my PRs today

File: ignite/engine.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ def add_event_handler(self, event_name, handler, *args, **kwargs):
 
         Parameters
         ----------
-        event_name: enum
-            event from ignite.trainer.TrainingEvents to attach the
+        event_name: Events
+            event from ignite.Engine.Events to attach the
```
Comment (alykhantejani) [inline ignite/engine.py]: Nice catch

File: ignite/handlers/timing.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    Args:
+        unit (bool, optional): if True, then when ``.value()`` method is called, the returned value
+            will be equal to total time measured, divided by the value of internal counter.
+
+    Attributes:
+        total (float): total time elapsed when the Timer was running (in seconds)
+        cnt (int): internal counter, usefull to measure average time, e.g. of processing a single batch.
+            Incremented with the ``.inc_unit()`` method.
+        running (bool): flag indicating if time is measuring time.
```
Comment (alykhantejani) [inline ignite/handlers/timing.py]: if **timer** is measuring time*

File: ignite/handlers/timing.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+            total += self._elapsed()
+
+        if self._unit:
+            denominator = max(self.cnt, 1.)
+        else:
+            denominator = 1.
+
+        return total / denominator
+
+    def inc_unit(self):
```
Comment (alykhantejani) [inline ignite/handlers/timing.py]: should we  rename this to `setp()` to be consistent with the docstring in `attach`?

File: ignite/handlers/timing.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        ...              pause=Events.ITERATION_COMPLETED,
+        ...              step=Events.ITERATION_COMPLETED)
+    """
+
+    def __init__(self, unit=False):
+        self._unit = unit
+        self._t0 = time.perf_counter()
+
+        self.total = 0.
+        self.cnt = 0.
```
Comment (alykhantejani) [inline ignite/handlers/timing.py]: maybe rename this to `step_count` to be clearer as its a public attribute

File: ignite/handlers/timing.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+        >>> trainer = Trainer(trainig_update_function)
+        >>> timer = Timer(unit=True)
+        >>> timer.attach(trainer,
+        ...              start=Events.EPOCH_STARTED,
+        ...              resume=Events.ITERATION_STARTED
+        ...              pause=Events.ITERATION_COMPLETED,
+        ...              step=Events.ITERATION_COMPLETED)
+    """
+
+    def __init__(self, unit=False):
```
Comment (alykhantejani) [inline ignite/handlers/timing.py]: should we rename `unit` to `mean` or `average`, I know `unit` is used in tnt but I still can't understand why

**Comment by elanmart** [comment]
I agree that the torchnet naming is a little bit unfortunate. I've updated the PR.

**Comment by alykhantejani** [comment]
Thanks @elanmart !

**Pull request merged**